src/main/java/org/testng/ITestRunnerFactory2.java

```diff
@@ -0,0 +1,28 @@
+package org.testng;
+
+import org.testng.xml.XmlTest;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A factory for TestRunners to be used by SuiteRunners.
+ */
+public interface ITestRunnerFactory2 extends ITestRunnerFactory {
```

Comment:
In fact, it won't work because we will never call the default method and should be removed at the end.

You can already wrap `ITestRunnerFactory` into an `ITestRunnerFactory2` (just make the transformation at `addListener` time) and it will allow us the manage `ITestRunnerFactory2` only.